app/src/main/java/com/gst/gusto/review/viewmodel/ReviewViewModel.kt

```diff
@@ -0,0 +1,119 @@
+package com.gst.gusto.review.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.gst.gusto.api.ApiResponse
+import com.gst.gusto.model.InstaReview
+import com.gst.gusto.repository.AuthRepositoryImpl
+import com.gst.gusto.repository.ReviewsRepositoryImpl
+import com.gst.gusto.util.GustoApplication
+import kotlinx.coroutines.launch
+
+class ReviewViewModel(
+    private val reviewsRepository: ReviewsRepositoryImpl,
+    private val authRepository: AuthRepositoryImpl
+) : ViewModel() {
+
+    private val xAuthToken = GustoApplication.prefs.getSharedPrefs().first
+    private val refreshToken = GustoApplication.prefs.getSharedPrefs().second
+
+    private val _instaReviews = MutableLiveData<MutableList<InstaReview?>>()
+    val instaReviews : LiveData<MutableList<InstaReview?>> = _instaReviews
+
+    private val _instaCursorId = MutableLiveData<Long>()
+    val instaCursorId : LiveData<Long> = _instaCursorId
+
+    private val _instaHasNext = MutableLiveData<Boolean>()
+    val instaHasNext : LiveData<Boolean> = _instaHasNext
+
+    private val _instaPagingData = MediatorLiveData<Pair<Long?, Boolean?>>()
+    val instaPagingdata : LiveData<Pair<Long?, Boolean?>> = _instaPagingData
+
+    private val _tokenToastData: MutableLiveData<Unit> = MutableLiveData()
+    val tokenToastData: LiveData<Unit> = _tokenToastData
+
+    private val _errorToastData: MutableLiveData<Unit> = MutableLiveData()
+    val errorToastData: LiveData<Unit> = _errorToastData
+
+    init {
+        viewModelScope.launch {
+            getInstaReview(18)
+
+            _instaPagingData.addSource(_instaCursorId) { cursorId ->
+                _instaPagingData.value = Pair(cursorId, _instaHasNext.value)
+            }
+
+            _instaPagingData.addSource(_instaHasNext) { hasNext ->
+                _instaPagingData.value = Pair(_instaCursorId.value, hasNext)
+            }
+        }
+
+    }
+
+    suspend fun getInstaReview(size: Int) {
```

Comment:
여기다가 함수 하나 만들까봐요.
똑같이는 안되겠지만 이런 형태로요 ~
```suggestion
    }

    fun onScrolled(lastCompletelyVisibleItemPosition: Int) {
        if (isFetching) return  // flag 대신 사용한 Boolean 타입 프로퍼티
        
        // 뷰모델 내에서는 LiveData 의 value 를 꺼내써도 괜찮아요.
        val lastIndex = _instaReview.value?.lastIndex.orZero()
        val hasNext = _instaPagingData.value?.second ?: false
        if (lastCompletelyVisiblePosition >= lastIndex && hasNext) {
            isFetching = true // 데이터 가져올때 true 로 변경
            // .. 여기서 이제 다음 데이터를 가져오는 작업을 해야겠죠.
            isFetching = false // 다 끝났으면 false 로 변경해주고 .. 
        }
    }

    suspend fun getInstaReview(size: Int) {
```